include/bluetooth/audio/vocs.h

```diff
@@ -0,0 +1,304 @@
+ */
+struct bt_vocs *bt_vocs_free_instance_get(void);
+
+/**
+ * @brief Get the service declaration attribute.
+ *
+ * The first service attribute returned can be included in any other GATT service.
+ *
+ * @param vocs Volume Offset Control Service instance.
+ *
+ * @return Pointer to the attributes of the service.
+ */
+void *bt_vocs_svc_decl_get(struct bt_vocs *vocs);
+
+/**
+ * @brief Initialize the Volume Offset Control Service instance.
+ *
+ * @param vocs      Volume Offset Control Service instance.
+ * @param init      Volume Offset Control Service initialization structure.
+ *                  May be NULL to use default values.
+ *
+ * @return 0 if success, ERRNO on failure.
+ */
+int bt_vocs_init(struct bt_vocs *vocs, const struct bt_vocs_init_param *init);
+
+/**
+ * @brief Callback function for the offset state.
+ *
+ * Called when the value is read, or if the value is changed by either the server or client.
+ *
+ * @param conn        Connection to peer device, or NULL if local server read.
+ * @param inst        The instance pointer.
+ * @param err         Error value. 0 on success, GATT error or ERRNO on fail.
+ *                    For notifications, this will always be 0.
+ * @param offset      The offset value.
+ */
+typedef void (*bt_vocs_state_cb_t)(
+	struct bt_conn *conn, struct bt_vocs *inst, int err, int16_t offset);
+
+/**
+ * @brief Callback function for setting offset.
+ *
+ * @param conn        Connection to peer device, or NULL if local server write.
+ * @param inst        The instance pointer.
+ * @param err         Error value. 0 on success, GATT error on fail.
+ */
+typedef void (*bt_vocs_set_offset_cb_t)(
+	struct bt_conn *conn, struct bt_vocs *inst, int err);
+
+/**
+ * @brief Callback function for the location.
+ *
+ * Called when the value is read, or if the value is changed by either the server or client.
+ *
+ * @param conn         Connection to peer device, or NULL if local server read.
+ * @param inst         The instance pointer.
+ * @param err          Error value. 0 on success, GATT error or ERRNO on fail.
+ *                     For notifications, this will always be 0.
+ * @param location     The location value.
+ */
+typedef void (*bt_vocs_location_cb_t)(
+	struct bt_conn *conn, struct bt_vocs *inst, int err, uint32_t location);
+
+/**
+ * @brief Callback function for the description.
+ *
+ * Called when the value is read, or if the value is changed by either the server or client.
+ *
+ * @param conn         Connection to peer device, or NULL if local server read.
+ * @param inst         The instance pointer.
+ * @param err          Error value. 0 on success, GATT error or ERRNO on fail.
+ *                     For notifications, this will always be 0.
+ * @param description  The description as an UTF-8 encoded string.
+ */
+typedef void (*bt_vocs_description_cb_t)(
+	struct bt_conn *conn, struct bt_vocs *inst, int err, char *description);
```

Comment:
I guess it's subjective, but I think it'd look nicer if you had included some function parameters on the first line, done the split after a parameter (i.e. after a `,` character). Same for the other function pointer typedefs

Reply:
Agreed. This was previously impossible due to the 80 line length (as some single parameters couldn't not fit on the line if starting at `(`, but it's possible now.